File: api-xmlsignature/src/tr/gov/tubitak/uekae/esya/api/xmlsignature/document/InMemoryDocument.java
package tr.gov.tubitak.uekae.esya.api.xmlsignature.document;

import tr.gov.tubitak.uekae.esya.api.xmlsignature.DataType;
import tr.gov.tubitak.uekae.esya.api.xmlsignature.XMLSignatureException;

import java.io.ByteArrayInputStream;
import java.io.InputStream;

/**
 * Document for in memory byte[]
 *
 * @author ahmety
 * date: May 14, 2009
 */
public class InMemoryDocument extends Document
{
    private byte[] mBytes;
    private InputStream mStream;

    public InMemoryDocument(byte[] aBytes, String aURI, String aMIMEType, String aEncoding)
    {
        super(aURI, aMIMEType, aEncoding);
        mBytes = aBytes;
    }

    public DataType getType()
    {
        return DataType.OCTETSTREAM;
    }

    @Override
    public byte[] getBytes() throws XMLSignatureException
    {
        return mBytes;
    }

    public InputStream getStream()
    {
        if (mStream==null){
            mStream = new ByteArrayInputStream(mBytes);
        }
        return mStream;
    }

}

File: api-xmlsignature/csharp/src/document/InMemoryDocument.cs
using System.IO;

namespace tr.gov.tubitak.uekae.esya.api.xmlsignature.document
{

	using DataType = tr.gov.tubitak.uekae.esya.api.xmlsignature.DataType;
	using XMLSignatureException = tr.gov.tubitak.uekae.esya.api.xmlsignature.XMLSignatureException;


	/// <summary>
	/// Document for in memory byte[]
	/// 
	/// @author ahmety
	/// date: May 14, 2009
	/// </summary>
	public class InMemoryDocument : Document
	{
		private readonly byte[] mBytes;
		private Stream mStream;

		public InMemoryDocument(byte[] aBytes, string aURI, string aMIMEType, string aEncoding) : base(aURI, aMIMEType, aEncoding)
		{
			mBytes = aBytes;
		}

		public override DataType Type
		{
			get
			{
				return DataType.OCTETSTREAM;
			}
		}

		public override byte[] Bytes
		{
			get
			{
				return mBytes;
			}
		}

		public override Stream Stream
		{
			get
			{
				if (mStream == null)
				{
					mStream = new MemoryStream(mBytes);
				}
				return mStream;
			}
		}

	}

}